include/nbl/builtin/hlsl/ieee754.hlsl

```diff
@@ -0,0 +1,128 @@
+#ifndef _NBL_BUILTIN_HLSL_IEE754_H_INCLUDED_
+#define _NBL_BUILTIN_HLSL_IEE754_H_INCLUDED_
+
+#include <nbl/builtin/hlsl/type_traits.hlsl>
+
+namespace nbl::hlsl::ieee754
+{
+	namespace impl
+	{
+		template <typename T>
+		unsigned_integer_of_size<sizeof(T)>::type castToUintType(T x)
+		{
+			using AsUint = typename unsigned_integer_of_size<sizeof(T)>::type;
+			return bit_cast<AsUint, T>(x);
+		}
+
+		// to avoid bit cast from uintN_t to uintN_t
+		template <> unsigned_integer_of_size<2>::type castToUintType(uint16_t x) { return x; }
+		template <> unsigned_integer_of_size<4>::type castToUintType(uint32_t x) { return x; }
+		template <> unsigned_integer_of_size<8>::type castToUintType(uint64_t x) { return x; }
+
+		template <typename T>
+		T replaceBiasedExponent(T x, uint32_t biasedExp)
+		{
+			static_assert(is_same<T, uint16_t>::value || is_same<T, uint32_t>::value || is_same<T, uint64_t>::value,
+				"Invalid type! Only floating point or unsigned integer types are allowed.");
+			return bitfield_insert(x, T(biasedExp), getMantissaBitCnt<T>(), getExponentBitCnt<T>());
+		}
+
+		template<>
+		float16_t replaceBiasedExponent<float16_t>(float16_t x, uint32_t biasedExp)
+		{
+			return bit_cast<float16_t, uint16_t>(replaceBiasedExponent<uint16_t>(bit_cast<uint16_t, float16_t>(x), biasedExp));
+		}
+
+		template<>
+		float32_t replaceBiasedExponent<float32_t>(float32_t x, uint32_t biasedExp)
+		{
+			return bit_cast<float32_t, uint32_t>(replaceBiasedExponent<uint32_t>(bit_cast<uint32_t, float32_t>(x), biasedExp));
+		}
+
+		template<>
+		float64_t replaceBiasedExponent(float64_t x, uint32_t biasedExp)
+		{
+			return bit_cast<float64_t, uint64_t>(replaceBiasedExponent<uint64_t>(bit_cast<uint64_t, float64_t>(x), biasedExp));
+		}
+	}
+
+	template <typename T>
+	int getExponentBitCnt() { return 0xdeadbeefu; }
```

Comment:
because DXC doesn't do `constexpr` functions, I'd rather declare a 
```cpp
template<typename Float>
struct float_traits
{
   NBL_CONSTEXPR_STATIC_INLINE exponentBitCnt = 0xdeadbeefu;
   ...etc...
};
```

Reply:
write done, don't resolve, and only after you psuh the commits that resolve